[blueiris] Blue Iris Binding initial contribution

pinkfish opened the pull request:
Ok, after hours...  I finally got the master branch sorted out and got this to apply a rebased change list that only has the blueiris changes in it.

This should be all that is needed for the blueiris plugin

Signed-off-by: David Bennett<pinkfishfrog@gmail.com> (github: pinkfish)

martinvw commented:
I see 6 files that should not be here, but can be resolved in this PR, thanks!

Note: this is a follow-up of #1983

martinvw commented on addons/binding/org.openhab.binding.nest.test/.project:
```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Can you remove this file from your PR

martinvw commented on addons/binding/org.openhab.binding.nest.test/.classpath:
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Can you remove this file from your PR

martinvw commented on addons/binding/org.openhab.binding.insteonplm/.project:
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Can you remove this file from your PR

martinvw commented on addons/binding/org.openhab.binding.elkm1/.project:
```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Can you remove this file from your PR

martinvw commented on addons/binding/org.openhab.binding.blueiris/pom.xml~Add tests to the right locations:
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Can you remove this file from your PR

martinvw commented on addons/binding/org.openhab.binding.blueiris/pom.xml~HEAD:
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Can you remove this file from your PR

martinvw commented on addons/binding/org.openhab.binding.blueiris/ESH-INF/i18n/blueiris_xx_XX.properties:
```diff
@@ -0,0 +1,11 @@
+# binding
```
Can you remove this file from your PR

martinvw commented on addons/binding/org.openhab.binding.blueiris/.project~HEAD:
```diff
@@ -0,0 +1,39 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Can you remove this file from your PR

martinvw commented on addons/binding/org.openhab.binding.blueiris/.project~Add tests to the right locations:
```diff
@@ -0,0 +1,39 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Can you remove this file from your PR

martinvw commented on addons/binding/org.openhab.binding.blueiris/OSGI-INF/BlueIrisHandlerFactory.xml:
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
```
It seems that this file has a missing, outdated or otherwise invalid copyright header, can you fix all of them by running 

`mvn license:format` 

Please commit only the ones for your binding.

kubawolanin commented:
@pinkfish please remember to [Sign-Off your commits](http://docs.openhab.org/developers/contributing/contributing#sign-your-work).

pinkfish commented:
Ok, updated based on the comments.  Rebasing again since it did annoying things and added all the bits back in again on this other computer.  Git is super frustrating at times :(  The name of the commit now wrong though and I cannot seem to edit at all on windows, since git commit --ammend is not working.

martinvw commented:
It is not signed-off :-(

pinkfish commented:
Ok, added in the signed off check and rebased (again) since apparently switching between computers resurrects the old history from git every time.  I think I need to clone from scratch on my other computer before I use it again to stop the history from resurrecting itself again.

martinvw commented:
[ERROR] ERROR found by checkstyle: .binding.blueiris/about.html:0 Invalid or missing license header in the about.html file. Here is an example of a valid about.html file: https://raw.githubusercontent.com/openhab/openhab2-addons/master/src/etc/about.html
[ERROR] ERROR found by checkstyle: .binding.blueiris/about.html:0 Invalid or missing license paragraph in the about.html file. Here is an example of a valid about.html file: https://raw.githubusercontent.com/openhab/openhab2-addons/master/src/etc/about.html
[ERROR] Detailed report can be found at: file////home/travis/build/openhab/openhab2-addons/addons/binding/org.openhab.binding.blueiris/target/code-analysis/report.html
[ERROR] -> [Help 1]

martinvw commented:
@pinkfish how are you doing? And what are your plans with respect to your open PR's.

Do you plan to work on them interleaved or do you prefer a certain order, if I can save some time now by not reviewing this one until we worked them down in your prefered order that would help out other people waiting for a review,

Please let me know.

pinkfish commented:
Ok, updated to fix the style issues with this patch.  All compiles and looks good now.

pinkfish commented:
Not super clear on where to put the signed off by.  Says it needs to put them into the commit but I am not sure I see a way to edit the commit message?

kaikreuzer commented:
Easily possible, see e.g. https://stackoverflow.com/a/20853093.

kaikreuzer commented:
@pinkfish Small git advice: Never ever do merge commits on your branch - do a "rebase on master" instead. This prevents a lot of headaches when trying to squash stuff later on :-)

pinkfish commented:
Ok, how do I get rid of the existing merge commits on the branch?  I rebased the main thing into one commit I think.

martinvw commented:
I expect that should work if you 

```
git rebase -i HEAD~3
```

And just remove the two commits:

**388e7df**8d6c6afe76f2f4e6919134848c45f165b
**c454014**42ddbcaa778908a0da66dda667b743f4f

Please check everything before you force (!) push to github.

martinvw commented:
@pinkfish any progress with the git magic here?

martinvw commented:
> @pinkfish any progress with the git magic here?

Hello, how are you doing? Do you have plans to finalize this binding? Or would you prefer someone else to take it over? In that case we could maybe tag it and ask other developers to jump in?

pinkfish commented:
I am fine with other people takin this over FYI

On Tue, May 29, 2018, 1:54 AM Martin van Wingerden <notifications@github.com>
wrote:

> @pinkfish <https://github.com/pinkfish> any progress with the git magic
> here?
>
> Hello, how are you doing? Do you have plans to finalize this binding? Or
> would you prefer someone else to take it over? In that case we could maybe
> tag it and ask other developers to jump in?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openhab/openhab2-addons/pull/2266#issuecomment-392702954>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACbmMXxKs-yzkA3FU06m0xbaWNYBgSExks5t3QzVgaJpZM4Natv9>
> .
>

Hilbrand commented:
Closing for now. There needs to done a lot of work to get this updated and if someone new wants to take over it's better to open a new pull request.
